libs/priors/bbhnet/priors/utils.py

```diff
@@ -3,8 +3,15 @@
 
 import h5py
 import numpy as np
-from bilby.core.prior import Interped
-from bilby.gw.prior import BBHPriorDict
+from bilby.core.prior import Interped, PriorDict
+
+
+def mass_ratio_constraint(samples):
+    if "mass_1" not in samples.keys() or "mass_2" not in samples.keys():
+        raise KeyError("mass_1 and mass_1 must exist to have a mass_ratio")
+    out_samples = samples
+    out_samples["mass_ratio"] = samples["mass_2"] / samples["mass_1"]
```

Comment:
Is there a reason for reassigning variable here? If so, I think you actually wan't `out_samples = samples.copy()`, what you've done here is basically add another reference to the dictionary data structure samples was pointing to, so altering `out_samples` will also alter `samples`

Reply:
Yeah, you're correct, should be `samples.copy()`

Reply:
Yeah if this is all happening internally to bilby we don't necessarily need to a do a copy here do we? Obviously not super important but just curious

Reply:
It ends up not mattering because `bilby`'s conversion function doesn't generate new parameters the way that  (I think) it's supposed to.